src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,223 @@
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._init_shared_b,
+            layer_normalization=self._layer_normalization
+        )
+
+    def _init_shared_b(self, b):
+        nn.init.zeros_(b[..., :self._action_dim])
+        nn.init.constant_(b[..., self._action_dim:], self._init_std_param.item())
+
+    def _get_mean_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+    def _get_adaptive_std_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._std_hidden_sizes,
+            hidden_nonlinearity=self._std_hidden_nonlinearity,
+            hidden_w_init=self._std_hidden_w_init,
+            hidden_b_init=self._std_hidden_b_init,
+            output_nonlinearity=self._std_output_nonlinearity,
+            output_w_init=self._std_output_w_init,
+            output_b_init=self._init_adaptive_b,
+            layer_normalization=self._layer_normalization)
+
+    def _init_adaptive_b(self, b):
+        return nn.init.constant_(b, self._init_std_param.item())
+
+    def forward(self, inputs):
+        if self._std_share_network:
+            output = self.shared_mean_std_network(inputs)
+            mean = output[..., :self._action_dim]
+            log_std = output[..., self._action_dim:]
+        else:
+            mean = self.mean_module(inputs)
+
+            if self._adaptive_std:
+                log_std = self.adaptive_std_module(inputs)
+            else:
+                broadcast_shape = list(inputs.shape[:-1]) + [self._action_dim]
+                x = torch.zeros(*broadcast_shape)
+                log_std = x + self._init_std_param
+
+        if self._min_std_param or self._max_std_param:
+            log_std = log_std.clamp(min=to_scalar(self._min_std_param), max=to_scalar(self._max_std_param))
+
+        if self._std_parameterization == 'exp':
+            log_std_var = log_std
+        else:
+            log_std_var = log_std.exp().exp().add(1.).log().log()
+
+        rnd = torch.rand(list(mean.shape)[1:])
+        action_var = rnd * log_std_var.exp() + mean
+
+        cov = (log_std_var.exp() ** 2).diag_embed()
+        dist = MultivariateNormal(mean, cov)
+
+        return action_var, mean, log_std_var, log_std, dist
```

Comment:
Please drop the `_var` postfix from `action_var` and `log_std_var`, since that postfix only really made sense in Tensorflow. It's confusing, since it might refer to "variance" but does not. The original `log_std` probably isn't needed.

Alternatively, you could rename `log_std` to `log_std_uncentered` or something.

Reply:
thanks! fixed!

Reply:
actually, please only return the dist (it should be sufficient for anyone's purposes). we are working on that refactoring in TF -- no need to add technical debt in torch.

Reply:
fixed

Reply:
only return dist, this dist contains mean and variance. Also we can get actions using rsample.